float64 speed
float64 default_speed
---
bool success

